CONTRIBUTING.md

```diff
@@ -0,0 +1,47 @@
+# Contributing to this package
+
+Let's build together! Please see our [Contributor Guide](https://docs.meltano.com/contribute/)
+for more information on contributing to Meltano.
+
+We believe that everyone can contribute and we welcome all contributions.
+
+Chat with us in [#contributing](https://meltano.slack.com/archives/C013Z450LCD) on [Slack](https://meltano.com/slack).
+
+Contributors are expected to follow our [Code of Conduct](https://docs.meltano.com/contribute/#code-of-conduct).
+
+## Semantic Pull Requests
+
+This repo uses the [semantic-prs](https://github.com/Ezard/semantic-prs) GitHub app to check all PRs againts the conventional commit syntax.
+
+Pull requests should be named according to the conventional commit syntax to streamline changelog and release notes management. We encourage (but do not require) the use of conventional commits in commit messages as well.
+
+In general, PR titles should follow the format "<type>: <desc>", where type is any one of these:
+
+- `ci`
+- `chore`
+- `build`
+- `docs`
+- `feat`
+- `fix`
+- `perf`
+- `refactor`
+- `revert`
+- `style`
+- `test`
+
+More advanced rules and settings can be found within the file [`.github/semantic.yml`](https://github.com/meltano/basic-python-template/blob/main/.github/semantic.yml).
+
+## Workspace Development Strategies for Meltano Python packages
+
+### Universal Code Formatting
+
+- From the [Black](https://black.readthedocs.io) website:
+    > By using Black, you agree to cede control over minutiae of hand-formatting. In return, Black gives you speed, determinism, and freedom from pycodestyle nagging about formatting. You will save time and mental energy for more important matters. **Black makes code review faster by producing the smallest diffs possible.** Blackened code looks the same regardless of the project you’re reading. **Formatting becomes transparent after a while and you can focus on the content instead.**
+
+### Pervasive Python Type Hints
+
+Type hints allow us to spend less time reading documentation. Public modules should be checked for type annotations on all methods and functions.
```

Comment:
Since this will only be used for new repositories, maybe we should take a stronger stance with regards to type hints by enforcing Mypy (in CI).

Reply:
👍 I'd be on board for that. 

@meltano/engineering anyone have any objections to that ?

Reply:
@WillDaSilva thoughts/prefs on a config? I've just got this config I used for a hackday project a while back:

```
[tool.mypy]
disallow_untyped_defs = true
check_untyped_defs = true
no_implicit_optional = true
warn_return_any = true
warn_unused_ignores = true
show_error_codes = true
```

But no idea whats considered a "good" mypy config.

Reply:
Looks good (and strict) to me. We should start with a strict config, and relax it as-needed on a per-module or per-import basis.